spec/controllers/books_controller_spec.rb

```diff
@@ -2,26 +2,43 @@
 
 RSpec.describe BooksController, type: :controller do
   let(:book){ create(:book) }
-  let(:books){ create_list(:book, 50) }
+  let(:book_list){ create_list(:book, 50) }
   let(:old_book){ create(:book, published_date: Date.today - 50.years)}
   let(:new_book){ create(:book, published_date: Date.today) }
 
   describe "GET #index" do
-    let(:send_request) { get :index }
+
+    let(:send_request){ get :index }
+    let(:sort_params){ { sort_field: "title", sort_order: "ASC" } }
+
     it "returns http success" do
       send_request
       expect(response).to have_http_status(:success)
     end
 
-    it "assigns all books to @books" do
-      send_request
-      expect(assigns(:books)).to eq(books)
+    context "when no params are sent" do
+      it "assigns all books to @books" do
+        send_request
+        expect(assigns(:books)).to eq(book_list)
+      end
+
+      it "sorts books by published_date" do
+        send_request
+        [old_book, new_book]
+        expect(assigns(:books)).to eq([new_book, old_book])
+      end
     end
 
-    it "sorts books by published_date" do
-      send_request
-      [new_book, old_book]
-      expect(assigns(:books)).to eq([old_book, new_book])
+    context "when a user searches" do
+      let(:send_request){ get :index, search_params }
+      let(:search_params){ { search: "This Unique Book" } }
+      let!(:book_list){ create_list(:book, 50) }
+      let!(:matching_book){ create(:book, title: "This Unique Book") }
+
+      it "should assign only matching books" do
```

Comment:
Avoid "should" in your test descriptions. Be declarative instead. `it "assigns only matching books" do`